File: Piramide/piramide.cpp
#include <iostream>
using namespace std;

void construirPiramide(int valor) {
    // Parte ascendente de la pirámide
    for (int i = 1; i <= valor; i++) {
        for (int j = 1; j <= i; j++) {
            cout << j << " ";
        }
        cout << endl;
    }

    // Parte descendente de la pirámide
    for (int i = valor - 1; i >= 1; i--) {
        for (int j = 1; j <= i; j++) {
            cout << j << " ";
        }
        cout << endl;
    }
}

int main() {
    int valor;
    char respuesta;

    do {
        cout << "Ingrese un valor: ";
        cin >> valor;

        // Llamamos a la función para construir la pirámide
        construirPiramide(valor);

        cout << "¿Desea construir otra pirámide? (s/n): ";
        cin >> respuesta;
    } while (respuesta == 's' || respuesta == 'S');

    return 0;
}

File: Piramide/piramide.py
def construir_piramide(valor):
    for i in range(1, valor + 1):
        for j in range(1, i + 1):
            print(j, end=" ")
        print()

    for i in range(valor - 1, 0, -1):
        for j in range(1, i + 1):
            print(j, end=" ")
        print()


def main():
    respuesta = 's'

    while respuesta[0].lower() == 's':
        valor = int(input('Ingresa un valor: '))
        construir_piramide(valor)

        respuesta = input('Desea construir otra piramide? (s/n): ')


if __name__ == "__main__":
    main()
